frontend/src/components/EditRecipe/EditRecipeForm.js

```diff
@@ -1,25 +1,72 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { withRouter } from 'react-router-dom';
-import { createRecipe } from '../../lib/apiClient';
+import Select from 'react-select';
+import TagsBar from '../TagsBar/TagsBar';
+import { createRecipe, getTags } from '../../lib/apiClient';
 import Footer from '../Footer/Footer';
 import './EditRecipeForm.css';
 
+const customStyles = {
+  control: provided => ({
+    ...provided,
+    height: 70,
+  }),
+  multiValueLabel: provided => ({
+    ...provided,
+    display: 'none',
+  }),
+  multiValueRemove: provided => ({
+    ...provided,
+    display: 'none',
+  }),
+
+};
+
 const EditRecipeForm = ({ history }) => {
   const [title, setTitle] = useState('');
   const [description, setDescription] = useState('');
   const [difficulty, setDifficulty] = useState('ADVANCED');
   const [duration, setDuration] = useState('30 mins');
   const [servings, setServings] = useState('2');
+  const [tags, setTags] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [selectedTagsWithId, setSelectedTagsWithId] = useState([]);
+
+  useEffect(() => {
+    getTags().then((data) => {
+      if (tags.length === 0) setTags(data);
+    });
+  }, [tags]);
```

Comment:
I think this will cause this to get called twice (or in an infinite loop if there are no tags in the database). This last line means that we run this function whenever `tags` changes, and we set `tags` inside the function. So it causes itself to be called again. It's not the worst thing in the world, but it's not great either.

Reply:
We could use a `refreshTags` state object that starts as `true`, and then check that inside the `useEffect` here, and set it to false when we refresh the tags. Then use `refreshTags` as the catalyst here on line 39. Like:

```
  useEffect(() => {
    if (refreshTags) {
      getTags().then((data) => {
        if (tags.length === 0) setTags(data);
      });
      setRefreshTags(false);
    }
  }, [refreshTags]);
```

Or something like that.